gitversion.yml

```diff
@@ -0,0 +1,55 @@
+assembly-versioning-scheme: MajorMinorPatch
+mode: ContinuousDelivery
+continuous-delivery-fallback-tag: ''
+next-version: 0.1.0
+branches: # https://gitversion.net/docs/learn/branching-strategies/gitflow/examples
```

Comment:
Why do we need `GitVersion` exactly? Was setting the version number via the `CHANGELOG.md` not enough?

Reply:
https://github.com/petabridge/Petabridge.App.Web/pull/108#discussion_r792694503 - I think I follow some of this, but I'd appreciate a more high-level explanation since I've not used `GitVersion` at all

Reply:
> Why do we need `GitVersion` exactly? Was setting the version number via the `CHANGELOG.md` not enough?

We don't manually set the version number inside the `CHANGELOG.md` file but instead create a placeholder for the next version using `## [vNext]`. The code below is in charge of updating the `CHANGELOG.md`:
```csharp
 Target RunChangelog => _ => _
        .OnlyWhenDynamic(()=> GitVersion.BranchName == ReleaseBranch)
        .Description("Updates the release notes and version number in the `ChangeLog.md`")
        .Executes(() =>
        {
            FinalizeChangelog(ChangelogFile, GitVersion.MajorMinorPatch, GitRepository);

            Git($"add {ChangelogFile}");
            Git($"commit -m \"Finalize {Path.GetFileName(ChangelogFile)} for {GitVersion.MajorMinorPatch}.\"");

            //To sign your commit
            //Git($"commit -S -m \"Finalize {Path.GetFileName(ChangelogFile)} for {vNext}.\"");

            Git($"tag -f {GitVersion.MajorMinorPatch}");
        });
```
`GitVersion.MajorMinorPatch` holds the next version number. The version number depends on the branch strategy. With the branch strategy, `GitVersion` will know which part to increment - that is Major, Minor or Patch!